torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -867,7 +867,7 @@ def _same_dtype_views(view1, view2):
     base_args = []
     other_args = []
     for i, inpt in enumerate(fwd_inputs):
-        if isinstance(inpt, Tensor):
+        if isinstance(inpt, Tensor) and not inpt.is_sparse:
```

Comment:
meta point:
- This PR looks like it is adding support for sparse tensors to torch.compile
- "adding sparse tensor support to torch.compile" likely needs a design doc and discussion because of the amount of complexity. the PT2 stack assumes it is working on regular strided Tensors (except for the case of torch_dispatch / torch_function Tensor subclasses, which Sparse tensors are not)
- One "easy" way to fix the tests is to just have Dynamo graph break if it sees a sparse tensor

Reply:
Not really adding support - just making some things that previously failed hard either trivially pass or fail less hard.

Reply:
This particular change seems problematic though. It probably would be better for AOTAutograd to raise an error if it receives a sparse tensor input. Not sure how this affects how many tests start passing though.

Reply:
FWIW, the `.is_sparse` check covers only sparse COO tensors but we have also sparse compressed tensors. Here, I suggest using `inpt.layout is torch.strided` instead of `inpt.is_sparse`.